libraries/lib-string-utils/FromChars.cpp

```diff
@@ -0,0 +1,190 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file FromChars.cpp
+ @brief Define functions to convert numeric types to string representation.
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#include "FromChars.h"
+
+#include "3party/fast_float.h"
+
+#include <limits>
+#include <algorithm>
+
+namespace
+{
+// Converting c to unsigned here helps to eliminate one check 
+unsigned digitToInt(char c) noexcept
+{
+   return unsigned(c - '0');
+}
+
+template <typename ResultType>
+FromCharsResult FastStringToInt(
+   const char* first, const char* last, ResultType& value,
+   bool isNegative) noexcept
+{
+   using UnsignedResultType = std::make_unsigned_t<ResultType>;
+
+   const auto availableBytes = last - first;
+
+   if (availableBytes <= 0)
+      return { first, std::errc::invalid_argument };
+
+   UnsignedResultType result = digitToInt(*first);
+
+   if (result > 10)
+      return { first, std::errc::invalid_argument };
+
+   constexpr auto maxSafeDigits = std::numeric_limits<ResultType>::digits10 - 1;
```

Comment:
I think you don't need `- 1`.  You treat the last safe digit in the second loop below instead of the first.  Not wrong, but is it too much caution?

Reply:
I want to keep the consistent overflow checks.

I.e. for uint8_t digits10 is 3, but 257 is an overflow. So we allow to parse "57" quickly but give more caution while parsing "157" because we expect that it can be "557"

Reply:
> I want to keep the consistent overflow checks.
> 
> I.e. for uint8_t digits10 is 3, but 257 is an overflow. 

Is it really 3, not 2?  See immediately under "Example" https://en.cppreference.com/w/cpp/types/numeric_limits/digits10

Reply:
Hmm, yes, you are completely right!